backend/api/views/TimeRecordView.py

```diff
@@ -0,0 +1,83 @@
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from api.models import TimeRecord, EventType, Group
+from api.serializers.TimeRecordSerializer import TimeRecordSerializer
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from django.db.models import Q, F, Value, Func, IntegerField
+from django.db.models.functions import Collate, Concat
+from rest_framework.exceptions import ValidationError
+from rest_framework import status
+from django.utils import timezone
+
+import logging
+
+logger = logging.getLogger('django')
+
+
+@extend_schema(tags=['Time Record'], request=TimeRecordSerializer)
+class TimeRecordViewSet(viewsets.ModelViewSet):
+    queryset = TimeRecord.objects.all()
+    serializer_class = TimeRecordSerializer
+    permission_classes = [IsAuthenticated]
+    http_method_names = ['get', 'post', 'delete', 'patch']
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        athlete_name = self.request.query_params.get('athlete_name', None)
+        event_type_id = self.request.query_params.get('event_type_id', None)
+        group_id = self.request.query_params.get('group_id', None)
+
+        if athlete_name:
+            # Add deterministic collation to full name (concatenation of first_name and last_name)
+            
+            queryset = queryset.annotate(full_name_deterministic=Collate(
+                Concat(F('athlete__first_name'), Value(' '), F('athlete__last_name')),
+                'und-x-icu'
+            ))
+            # Add deterministic collation to first_name and last_name
+            queryset = queryset.annotate(first_name_deterministic=Collate("athlete__first_name", "und-x-icu"), last_name_deterministic=Collate("athlete__last_name", "und-x-icu"))
+            queryset = queryset.filter(
+                Q(full_name_deterministic__istartswith=athlete_name)|
```

Comment:
For `full_name` we are using `istartswith`, but for first name and last name` startswith`. 
Could you change it to` istartswith` for all, please?

Reply:
Sure, I missed that when putting them back, I just update the code.